src/arrayMethodJoin.js

```diff
@@ -4,8 +4,33 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(...args) {
+    const separator = args;
+    let result = '';
+
+    for (let i = 0; i < args.length; i++) {
+      if (separator[i] === undefined) {
+        separator[i] = ',';
+      } else if (separator[i] === null) {
+        separator[i] = 'null';
+      }
+    }
```

Comment:
```suggestion
    let result = '';
```
you don't need this just give default value to separator